tests/test_spidermiddleware_urllength.py

```diff
@@ -39,3 +40,14 @@ def test_logging(self):
         self.assertEqual(ric, 1)
 
         self.assertIn(f'Ignoring link (url length > {self.maxlength})', str(log))
+
+    def test_setting(self):
+        length = 10
+        settings = Settings({'URLLENGTH_LIMIT': length})
+        test_ok = UrlLengthMiddleware.from_settings(settings)
+        settings_error = Settings({'URLLENGTH_LIMIT': 0})
+        with self.assertRaises(NotConfigured) as cm:
+            UrlLengthMiddleware.from_settings(settings_error)
+        the_exception = cm.exception
+        self.assertEquals(type(the_exception), type(NotConfigured()))
+        self.assertEquals(test_ok.maxlength, length)
```

Comment:
Could you split every check that makes sense on its own into a separate test method?

Reply:
ok, it's on [this commit](https://github.com/scrapy/scrapy/pull/5158/commits/a4b768d419680c2125d9b567515f214585ae050b)